src/comsock.c

```diff
@@ -439,9 +440,11 @@ natsStatus
 natsSock_WriteFully(natsSockCtx *ctx, const char *data, int len)
 {
     natsStatus  s     = NATS_OK;
-    int         bytes = 0;
     int         n     = 0;
 
+    if (len == 0)
```

Comment:
How did a natsSock_Write of 0 behave previously?  return 0?

Reply:
Fairly confident that I was never calling with len==0, but otherwise, connection would probably have been reported as closed.